experts-client/lib/experts-client.js

```diff
@@ -111,31 +113,111 @@ export class ExpertsClient {
       "harvest_iam": "http://iam.ucdavis.edu/"
     };
     this.jsonld = '{"@context":' + JSON.stringify(context) + ',"@id":"http://iam.ucdavis.edu/", "@graph":' + JSON.stringify(docObj) + '}';
-    
+
   }
-  
+
   /**
    * create fuseki dataset
    */
   async createDataset(dbName, dbType) {
     try {
       await fetch( this.cli.iamEndpoint + '/$/datasets?dbName=' + dbName + '&dbType=' + dbType, {
-      method: 'POST',
-      body: '[]',
-      headers: { 
-        'Content-Type': 'application/json',
-        'Authorization': 'Basic ' + this.cli.fusekiAuth
-      }
-    }).then(res => res.text())
-    .catch(err => console.log(err));
-  } catch (err) {
+        method: 'POST',
+        body: '[]',
+        headers: {
+          'Content-Type': 'application/json',
+          'Authorization': 'Basic ' + this.cli.fusekiAuth
+        }
+      }).then(res => res.text())
+        .catch(err => console.log(err));
+    } catch (err) {
       console.log(err);
+    }
+  }
+
+  /**
+   * This could easily be joined w/ createDataset, and called mkDb and we only specify temp id if we done't have a name
+   **/
+  async mkFusekiTmpDb(opt,files) {
+    const fuseki=opt.fuseki;
+    if (!fuseki.url) {
+      throw new Error('No Fuseki url specified');
+    }
+    if (!fuseki.auth) {
```

Comment:
You can see that as a rule, I don't ever try and catch errors, I just throw those out of the library.  Not sure if this is compatible w/ Justin's SOP.